app/models/authentication.rb

```diff
@@ -81,6 +81,18 @@ def raise_event(status, _message = nil)
     MiqEvent.raise_evm_event_queue(ci, "#{prefix}_auth_#{status}")
   end
 
+  def assign_values(options)
```

Comment:
@Fryguy I think this is being used as a baseclass implementation and overridden in (at least one of the) subclasses.